netbox/extras/conditions.py

```diff
@@ -135,23 +135,24 @@ class ConditionSet:
     def __init__(self, ruleset):
         if type(ruleset) is not dict:
             raise ValueError(_("Ruleset must be a dictionary, not {ruleset}.").format(ruleset=type(ruleset)))
-        if len(ruleset) != 1:
-            raise ValueError(_("Ruleset must have exactly one logical operator (found {ruleset})").format(
-                ruleset=len(ruleset)))
-
-        # Determine the logic type
-        logic = list(ruleset.keys())[0]
-        if type(logic) is not str or logic.lower() not in (AND, OR):
-            raise ValueError(_("Invalid logic type: {logic} (must be '{op_and}' or '{op_or}')").format(
-                logic=logic, op_and=AND, op_or=OR
-            ))
-        self.logic = logic.lower()
 
-        # Compile the set of Conditions
-        self.conditions = [
-            ConditionSet(rule) if is_ruleset(rule) else Condition(**rule)
-            for rule in ruleset[self.logic]
-        ]
+        if len(ruleset) == 1:
```

Comment:
Is this the correct order below - it seems like they are reversed, the and-or processing is being done if the len(ruleset) == 1 which seems like it should be in the else clause?

Reply:
It's in the correct order because of the ruleset below, for example, has length = 1 and this case is deal inside the if. All other conditions must be deal in the else. Does it make sense?
```
{
    "or": [
        {
            "and": [
                {
                    "attr": "status.value",
                    "value": "active"
                },
                {
                    "attr": "primary_ip4",
                    "negate": true,
                    "value": null
                }
            ]
        },
        {
            "attr": "tags.slug",
            "op": "contains",
            "value": "exempt"
        }
    ]
}
```